src/app/component/ProductCard/ProductCard.component.js

```diff
@@ -35,25 +37,36 @@ class ProductCard extends Component {
         this.handleConfigurableClick = this.handleConfigurableClick.bind(this);
     }
 
-    getCurrentVariantIndex() {
-        const { product: { variants }, customFilters } = this.props;
-        const customFiltersExist = customFilters && Object.keys(customFilters).length;
+    getConfigurableParameters() {
+        const { product: { variants = [] }, customFilters = {} } = this.props;
+        const filterKeys = Object.keys(customFilters);
 
-        if (variants && customFiltersExist) {
-            for (let i = 0; i < variants.length; i++) {
-                const { product } = variants[ i ];
+        if (filterKeys.length < 0) return { indexes: [], parameters: {} };
 
-                const isCorrectVariant = Object.keys(customFilters).every(filterKey => (
-                    customFilters[ filterKey ].find(value => +value === +product[ filterKey ])
-                ));
+        const indexes = getVariantsIndexes(variants, customFilters);
+        const [index] = indexes;
 
-                if (isCorrectVariant) return i;
-            }
+        if (!variants[index]) return { indexes: [], parameters: {} };
```

Comment:
You could merge `filterKeys.length < 0` and `!variants[index]` checks together? It will appear more logical.
However in terms of performance your solution is better. You can leave current.